datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -269,7 +271,9 @@ def _submit_metric(self, mtype, name, value, tags=None, hostname=None, device_na
             self.warning(err_msg)
             return
 
-        aggregator.submit_metric(self, self.check_id, mtype, self._format_namespace(name), value, tags, hostname)
+        aggregator.submit_metric(
+            self, self.check_id, mtype, self._format_namespace(name), value, tags, hostname, device_name
+        )
```

Comment:
Note: `aggregator.submit_metric` signature cannot be changed, it should match the agent implementation.

Reply:
Removed device_name from base check